tests/type_inference/test_functions.py

```diff
@@ -30,12 +30,51 @@ def test_function_def_call_no_args(function_name, return_value):
     # there should be a single Expr node in this program
     expr_node = next(module.nodes_of_class(astroid.Expr))
     assert expr_node.type_constraints.type == type(return_value)
 
 
+@given(cs.valid_identifier(), hs.lists(cs.valid_identifier(), min_size=1))
+def test_function_def_args_simple_return(function_name, arguments):
+    """Test FunctionDef node visitors representing function definitions with params; return one of it's arguments."""
+    assume(not iskeyword(function_name) and function_name not in arguments)
+    # generate every possible function definition program of aforementioned form.
+    for i in range(len(arguments)):
+        return_value = arguments[i]
+        program = _parse_to_function(function_name, arguments, return_value)
+        module = cs._parse_text(program)
+        # get the functionDef node - there is only one in this test case.
+        function_def_node = [n for n in module.nodes_of_class(astroid.FunctionDef)][0]
+        expected_arg_type_vars = [function_def_node.type_environment.lookup_in_env(argument) for argument in arguments]
+        expected_arg_types = [TYPE_CONSTRAINTS.lookup_concrete(type_var) for type_var in expected_arg_type_vars]
+        function_type_var = module.type_environment.lookup_in_env(function_name)
+        function_type = TYPE_CONSTRAINTS.lookup_concrete(function_type_var)
+        actual_arg_types, actual_return_type = TYPE_CONSTRAINTS.types_in_callable(function_type)[0],\
+                             TYPE_CONSTRAINTS.types_in_callable(function_type)[1]
+        return_type_var = function_def_node.type_environment.lookup_in_env(return_value)
+        expected_return_type = TYPE_CONSTRAINTS.lookup_concrete(return_type_var)
+        assert Callable[actual_arg_types, actual_return_type] == Callable[expected_arg_types, expected_return_type]
+
+
+@given(cs.valid_identifier(), cs.random_dict_variable_value(min_size=1))
+def test_function_def_args_simple_function_call(function_name, variables_dict):
+    """Test setting f env for function call of function definitions with params that returns one of it's arguments."""
+    assume(not iskeyword(function_name) and function_name not in variables_dict)
+    # generate every possible function definition program of aforementioned form.
+    for i in range(len(list(variables_dict.keys()))):
+        return_value = list(variables_dict.keys())[i]
+        arguments = ", ".join([repr(value) for value in variables_dict.values()])
+        program = f'{_parse_to_function(function_name, list(variables_dict.keys()), return_value)}\n' \
+                  f'{function_name}({arguments})'
+        module = cs._parse_text(program)
+        # get the Call node - there is only one in this test case.
+        call_node = [n for n in module.nodes_of_class(astroid.Call)][0]
+        function_call_type = call_node.type_constraints.type
+        assert TYPE_CONSTRAINTS.lookup_concrete(function_call_type) == type(list(variables_dict.values())[i])
```

Comment:
Do not call `type` here. Instead, you should go into the `Arguments` child of the function call, and then access the `type_constraints.type` from the `i`-th argument node.